tools/linter/golangci-lint/.golangci.yml

```diff
@@ -4,18 +4,24 @@ run:
 linters:
   enable:
     - bodyclose
+    - depguard
+    - exportloopref
     - gofmt
     - goimports
-    - revive
+    - goheader
+    - gocritic
     - gosec
     - misspell
-    - exportloopref
+    - revive
     - unconvert
     - unparam
-    - goheader
-    - gocritic
 
 linters-settings:
+  depguard:
```

Comment:
is this needed, I dont think we use `gogo` in this project ?
and reg `gopkg.in/yaml.v3` - we actually need to use this package in https://github.com/envoyproxy/gateway/pull/893

Reply:
prevent use `gogo` in the future.

Reply:
updated with `nolint` for special usecase.